### ISSUE TITLE
Possible to have two items visibly selected

### ISSUE THREAD

**Issue by edwardhorsford** [alphagov/accessible-typeahead#74]
It's possible to have two items visibly selected - only one should be selected at any one time.

Steps to reproduce:
* Have a selected menu item
* Hover down to another item and leave mouse in place
* Use arrow keys to move selected item

Example:
![location-picker-bug-multiple-selected](https://cloud.githubusercontent.com/assets/2204224/25482687/bd68d528-2b4a-11e7-95c8-00e139fc8f2d.gif)

A possible _different_ solution to this might be to style 'hover-selected' items differently than the 'actually-selected' items.

**Comment by adamsilver** [comment]
> A possible different solution to this might be to style 'hover-selected' items differently than the 'actually-selected' items.

Good move, I reckon. We should try and differentiate those.

**Comment by tvararu** [comment]
> A possible different solution to this might be to style 'hover-selected' items differently than the 'actually-selected' items.

I think I suggested this but we didn't go for it because of the additional cognitive overhead. I think it would be a sound change and significantly simplify the code. 👍

**Comment by edwardhorsford** [comment]
@tvararu Is it something I can easily try? My *hunch* is that it would be better to avoid two selected states - but I'm willing to be proven wrong.

**Comment by tvararu** [comment]
@edwardhorsford yes! That solution is a CSS only change that involves playing with [these two classes](https://github.com/alphagov/accessible-typeahead/blob/ecf3668c3bc01532d0c5240c773da2a97ede188d/examples/styled.css#L85-L86).

**Comment by adamsilver** [comment]
Did you just try and break the component, or do we think there are users that will try and navigate the options by keyboard and mouse at the same time?

**Comment by tvararu** [comment]
@adamsilver it's actually easy to stumble upon this bug, if your mouse is somewhere in the area where the suggestions will appear and you arrow down then more often than not two things will be visually selected.

I don't think it's high priority but it is a valid bug.

**Comment by joelanman** [comment]
suggestion: just highlight the latest. Mouse hover then keyboard - just highlight the keyboard. Keyboard then mouse - just highlight the mouse. Think this is consistent with normal OS usage.

**Comment by tvararu** [comment]
@joelanman that's the original intention! Just that `:hover` combined with lack of an `not adjacent to` selector makes it needlessly complex. :)
